### ISSUE TITLE
VCS packages sometimes wrongly considered up to date as of 4.2.16

### ISSUE THREAD

**Issue by pmattern** [rmarquis/pacaur#316]
As of 4.2.16 pacaur sometimes considers VCS packages as up-to-date besides they aren't.

Example:
4.2.20, but same behaviour with all versions ≥ 4.2.16
```
$ pacaur -S sddm-git 
:: Package(s) sddm-git not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...
:: sddm-git latest revision -- fetching

AUR Packages  (1): sddm-git-latest  

:: Proceed with installation? [Y/n] 

:: Retrieving package(s)...
:: View sddm-git PKGBUILD? [Y/n] n
:: View sddm.install script? [Y/n] n
:: Installing sddm-git cached package...
loading packages...
warning: sddm-git-0.10.0.33.gc3ea723-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) sddm-git-0.10.0.33.gc3ea723-1

Total Installed Size:  5.70 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                                       [################################################] 100%
(1/1) checking package integrity                                                     [################################################] 100%
(1/1) loading package files                                                          [################################################] 100%
(1/1) checking for file conflicts                                                    [################################################] 100%
(1/1) checking available disk space                                                  [################################################] 100%
(1/1) reinstalling sddm-git
```
after downgrading to 4.2.15, no further modifications, in particular same configuration
```
$ pacaur -S sddm-git 
:: Package(s) sddm-git not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...
:: sddm-git latest revision -- fetching

AUR Packages  (1): sddm-git-latest  

:: Proceed with installation? [Y/n] 

:: Retrieving package(s)...
:: View sddm-git PKGBUILD? [Y/n] n
:: View sddm.install script? [Y/n] n
==> Making package: sddm-git 0.10.0.33.gc3ea723-1 (Wed Mar  4 22:10:32 CET 2015)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
[...]
```

Doesn't happen with all VCS packages and I couldn't track down the exact conditions when it takes place.
But whenever it happened downgrading to 4.2.15 prevented the issue. Thus I think pacaur must be involved somehow.

Only relevant configuration settings were PKGDEST in makepkg.conf as well as builddir and clean{,devel}=false in pacaur's config.

**Comment by rmarquis** [comment]
Thank a lot for the detailed bug report!

I did some improvement in 4.2.16 regarding the integrity check, since PGP integrity is basically enforced with pacman 4.2, and I think commit 91a9e18fa9c785e54f916e2a31c7061faeedfcb2 is responsible for that bug. If the package is already built and found in cache, then the pkgver() function is never run and pacaur won't know about newer VCS version.

**Comment by rmarquis** [comment]
Should be fixed in ce30704666f4a4a6e672ddf0af9937a643253d5c. You can try it by installing pacaur-git.

Note the package still won't be rebuild if the cached package is found up-to-date after comparing with latest upstream version, unless you use the --rebuild flag.

**Comment by pmattern** [comment]
After testing thoroughly for some days now the problem is gone. So thanks for the quick fix.
Unfortunately there seems to be one weird remnant:

The problem persists if one of the packages qupzilla-git or qupzilla-qt5-qtwebkit-git gets installed the first time with switch --needed set:
```
$ pacaur --needed -S qupzilla-git 
:: Package(s) qupzilla-git not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (1): qupzilla-git-latest  

:: Proceed with installation? [Y/n] 

:: Retrieving package(s)...
:: Checking qupzilla-git integrity...
==> Making package: qupzilla-git v1.8.0.r142.g8d7cfb0-1 (Mon Mar  9 15:59:53 CET 2015)
==> WARNING: Skipping dependency checks.
==> Retrieving sources...
  -> Cloning qupzilla git repo...
Cloning into bare repository '/usr/local/peter/Pakete/qupzilla-git/qupzilla'...
remote: Counting objects: 45383, done.
remote: Total 45383 (delta 0), reused 0 (delta 0), pack-reused 45383
Receiving objects: 100% (45383/45383), 43.59 MiB | 778.00 KiB/s, done.
Resolving deltas: 100% (35459/35459), done.
Checking connectivity... done.
==> Validating source files with md5sums...
    qupzilla ... Skipped
==> Extracting sources...
  -> Creating working copy of qupzilla git repo...
Cloning into 'qupzilla'...
done.
Switched to a new branch 'makepkg'       (*)
==> Starting pkgver()...
==> Sources are ready.
:: qupzilla-git is up-to-date -- skipping
```
Configuration as stated in first post, neither subdirectories $builddir/qupzilla-* nor any other checkout available when invoking pacaur.
First I thought it could be related to that strange branch "makepkg" (*). But the other day I came across a branch like this compiling some kernel package on Arch Linux ARM which works flawlessly...
So honestly I have no clue what's going on here. Maybe it's because both these packages are using fragments to download the source (couldn't find other packages doing the same at a glance)?

**Comment by rmarquis** [comment]
I can reproduce, reopening.

**Comment by rmarquis** [comment]
Fixed in d4e7413d401973e32a05678fa975063efd618373.
